### PR TITLE
DateTime adapter

### PR DESCRIPTION
**Pull request by Gustl22**
For lighting and shadow I want to add a tool which calculates the light direction of current day/time.
Unfortunately GWT doesn't support `java.util.Calendar`.
So here's an adapter to have access to the date on all platforms.

_FYI: I tested the gwt implementation via browser, but didn't manage to run a JUnit test._

If you want, can review and keep on separate branch, so can make it available for shadow implementation.

### CODE REVIEW

**Comment by devemux86** [comment]
Related to #575.

**Comment by devemux86** [comment]
Merged in branch [issue_575](https://github.com/mapsforge/vtm/tree/issue_575) for now, until we see its actual use.

**Comment by Gustl22** [comment]
I noticed, that I need the time zone, too. Here in the second commit. Blame me for force push, but I thought a second PR on same issue side branch is overkill :see_no_evil:

**Comment by devemux86** [comment]
There are not any PR submitted in [issue_575](https://github.com/mapsforge/vtm/tree/issue_575) branch.

File: vtm/src/org/oscim/backend/DateTime.java [hunk 1]
```diff
@@ -37,4 +37,10 @@ public int getSecond() {
     public int getDayOfYear() {
         return Calendar.getInstance().get(Calendar.DAY_OF_YEAR);
     }
+
+    @Override
+    public int getTimeZoneOffset() {
+        Calendar inst = Calendar.getInstance();
+        return inst.getTimeZone().getOffset(inst.getTimeInMillis()) / 60000;
```
Comment (devemux86) [inline vtm/src/org/oscim/backend/DateTime.java]: Better use the found milliseconds on supported platforms. And let apps decide on precision?

**Comment by devemux86** [comment]
Let's not review again PR once merged in branches, so better post PR in their branches for any improvements.

For now and since they're small changes I can copy / paste them squashed in branch [issue_575](https://github.com/mapsforge/vtm/tree/issue_575).

File: vtm/src/org/oscim/backend/DateTime.java (hunk 1 above)
Comment (Gustl22) [reply vtm/src/org/oscim/backend/DateTime.java]: Ok your choice :wink:

**Comment by Gustl22** [comment]
Yes next time I'll post separate PRs :)

**Comment by devemux86** [comment]
Closing since it's merged in its relevant branch.

**Comment by devemux86** [comment]
Thanks, merged via 0a8e56ea05159723a9d71e6b774192e3c2c97eae.

**Pull request closed without merging**